src/Compilers/CSharp/Portable/Declarations/DeclarationTreeBuilder.cs

```diff
@@ -619,7 +682,60 @@ public override SingleNamespaceOrTypeDeclaration VisitEnumDeclaration(EnumDeclar
                 nameLocation: new SourceLocation(node.Identifier),
                 memberNames: memberNames,
                 children: ImmutableArray<SingleTypeDeclaration>.Empty,
-                diagnostics: diagnostics.ToReadOnlyAndFree());
+                diagnostics: diagnostics.ToReadOnlyAndFree(),
+                _nonGlobalAliasedQuickAttributes | quickAttributes);
+        }
+
+        private static QuickAttributes GetQuickAttributes(SyntaxList<AttributeListSyntax> attributeLists)
+        {
+            var result = QuickAttributes.None;
+            foreach (var attributeList in attributeLists)
+            {
+                foreach (var attribute in attributeList.Attributes)
+                    result |= GetQuickAttributes(attribute.Name.GetUnqualifiedName().Identifier.ValueText, inAttribute: true);
+            }
+
+            return result;
+        }
+
+        private static QuickAttributes GetQuickAttributes(string name, bool inAttribute)
+        {
+            var result = QuickAttributes.None;
+            if (Matches(AttributeDescription.TypeIdentifierAttribute))
+                result |= QuickAttributes.TypeIdentifier;
+
+            if (Matches(AttributeDescription.TypeForwardedToAttribute))
```

Comment:
sure. these could def be else-if.  

> At a glance it looks like only TypeIdentifierAttribute really requires searching the type declarations--the rest can only be applied to assemblies.

i didn't want to make presumptions about that as it felt like an easy way for issues to sneak in where we missed a case here but didn't have an issue with it because the code that produces all attributes for a symbol often may run prior to a direct call to look for a particular attribute.